core/src/main/java/org/elasticsearch/action/admin/indices/create/TransportCreateIndexAction.java

```diff
@@ -83,4 +83,9 @@ protected void masterOperation(final CreateIndexRequest request, final ClusterSt
             listener::onFailure));
     }
 
+
+    @Override
+    protected void validateRequest(CreateIndexRequest request, ClusterState state) throws Exception {
+        MetaDataCreateIndexService.validateIndexName(request.index(), state);
```

Comment:
I see where you're heading with the extra state validation, but name collisions are a rare occasion and I'm not sure it's worth it to have the extra API complexity in order to fail early. If we remove that we can move the name validation to `org.elasticsearch.action.admin.indices.create.CreateIndexRequest#validate`

Reply:
The lack of validation in one of the calls actually caused behavior to break between major versions of ES. I think we should avoid this, but I am completely fine w keeping it out of a generic place.